src/domain/data-decoder/v2/entities/data-decoded.entity.spec.ts

```diff
@@ -0,0 +1,355 @@
+        },
+      ]);
+    });
+
+    it('should expect a numeric string value', () => {
+      const multisend = multisendBuilder()
+        .with('value', faker.string.hexadecimal())
+        .build();
+
+      const result = MultisendSchema.safeParse(multisend);
+
+      expect(!result.success && result.error.issues).toStrictEqual([
+        {
+          code: 'custom',
+          message: 'Invalid base-10 numeric string',
+          path: ['value'],
+        },
+      ]);
+    });
+
+    it('should checksum the to', () => {
+      const nonChecksummedAddress = faker.finance
+        .ethereumAddress()
+        .toLowerCase();
+      const multisend = multisendBuilder()
+        .with('to', nonChecksummedAddress as `0x${string}`)
+        .build();
+
+      const result = MultisendSchema.safeParse(multisend);
+
+      expect(result.success && result.data.to).toBe(
+        getAddress(nonChecksummedAddress),
+      );
+    });
+
+    it('should expect hex data', () => {
+      const multisend = multisendBuilder()
+        .with('data', faker.string.alpha() as `0x${string}`)
+        .build();
+
+      const result = MultisendSchema.safeParse(multisend);
+
+      expect(!result.success && result.error.issues).toStrictEqual([
+        {
+          code: 'custom',
+          message: 'Invalid "0x" notated hex string',
+          path: ['data'],
+        },
+      ]);
+    });
+
+    it('should not validate an invalid Multisend', () => {
+      const multisend = { invalid: 'multisend' };
+
+      const result = MultisendSchema.safeParse(multisend);
+
+      expect(!result.success && result.error.issues).toStrictEqual([
+        {
+          code: 'invalid_type',
+          expected: '0 | 1',
+          message: 'Required',
+          path: ['operation'],
+          received: 'undefined',
+        },
+        {
+          code: 'invalid_type',
+          expected: 'string',
+          message: 'Required',
+          path: ['value'],
+          received: 'undefined',
+        },
+        {
+          code: 'invalid_type',
+          expected: 'object',
+          message: 'Required',
+          path: ['dataDecoded'],
+          received: 'undefined',
+        },
+        {
+          code: 'invalid_type',
+          expected: 'string',
+          message: 'Required',
+          path: ['to'],
+          received: 'undefined',
+        },
+        {
+          code: 'invalid_type',
+          expected: 'string',
+          message: 'Required',
+          path: ['data'],
+          received: 'undefined',
+        },
+      ]);
+    });
+  });
+
+  describe('ValueDecodedSchema', () => {
+    it('should validate a Multisend', () => {
+      const multisend = multisendBuilder().build();
+
+      const result = ValueDecodedSchema.safeParse([multisend]);
+
+      expect(result.success).toBe(true);
+    });
+
+    it('should validate a BaseDataDecoded', () => {
+      const baseDataDecoded = baseDataDecodedBuilder().build();
```

Comment:
Should we add a test case with several parameters? That case is not considered in the builder to avoid a call stack exceed, but it can be the case on "real" scenarios, right?

Reply:
Added in 01ea3dcd9a518d3da2cec7409906d9a758b3ac52,